bokeh/document.py

```diff
@@ -336,6 +350,10 @@ def to_json_string(self):
             }
         }
 
+        custom_models = self._collect_custom_models()
+        if custom_models:
+            json['models'] = custom_models
+
```

Comment:
I think this will cause us problems; models code is not part of Document. Here we are tacking an extra payload on to Document that we'd like to go to JavaScript, because we happen to know we'll send the Document to JavaScript.
It's an unexpected side effect that `Document.from_json` on the coffee side loads custom models, it should only fill in `Document`, not have surprising side effects.

What I think would be cleaner: in `_standalone_docs_json_and_render_items`, redefine `docs_json` to include the models (maybe demote the current dict of docs, so it's `{ 'models' : { ... }, 'docs' : current_docs_json_here }` and then in `Bokeh.embed.embed_items` of course the code has to be changed to match.